### PR TITLE
[ios] Experimental Expo modules support for Fabric

### PR DESCRIPTION
**Pull request by tsapeta**
# Why

The goal is to support both Paper and Fabric in Expo modules written with the new API.
Ref ENG-4738

# How

- Created a separate repo for the simple example app using Fabric and `expo-linear-gradient`
- Added this example app as a submodule in `apps` folder
- Implemented Fabric-specific C++ classes, usually generated by codegen
- Added `React-RCTFabric` and `RCT-Folly` as dependencies of `expo-modules-core`, but only if `RCT_NEW_ARCH_ENABLED` flag is set
- Added `ExpoFabricView` that wraps the view provided by the module. Unfortunately, Fabric creates views on its own, so we cannot pass anything to the view instance. As a workaround, we're dynamically creating a subclass of `ExpoFabricView` for each view module that has the `AppContext` and the associated module name injected.
- Fixed BlurView animation (made by @kudo), but since we probably won't add support for Fabric in legacy modules, maybe there is no reason to include this fix?

# Test Plan

Used the fabric-tester app that demonstrates mounting/unmounting and updating the props in `expo-linear-gradient`

https://user-images.githubusercontent.com/1714764/182871087-3c30ce9c-c825-4f25-81be-c50a17b139fb.mp4

### CODE REVIEW

**Review by Kudo — APPROVED** [review]
awesome work 🔥

**Comment by tsapeta** [comment]
*Hi there! 👋 I'm a bot whose goal is to ensure your contributions meet our guidelines.*

I've found some issues in your pull request that should be addressed (click on them for more details) 👇

<details>
  <summary><strong>⚠️ Suggestion</strong>: SwiftLint Violations</summary>

&NewLine;
Found 2 violations, 0 serious. See the review comments below for more insights.
</details>


---
*Generated by ExpoBot 🤖 against ed6f85edeb7a1621540ac83e0108aab588bd4b34*

File: packages/expo-modules-core/ios/Fabric/ExpoFabricView.swift [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  // MARK: - ExpoFabricViewInterface
+
+  public override func updateProp(_ propName: String, withValue value: Any) {
+    guard let view = contentView else {
+      return
+    }
+    if let _ = moduleHolder, let prop = viewManagerPropDict?[propName] {
```
Comment (tsapeta) [inline packages/expo-modules-core/ios/Fabric/ExpoFabricView.swift:58]: #### 🟠 SwiftLint: [Unused Optional Binding Violation](https://realm.github.io/SwiftLint/unused_optional_binding.html)
Prefer `!= nil` over `let _ =`

File: packages/expo-modules-core/ios/Fabric/ExpoFabricView.swift [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+      // TODO: @tsapeta: Figure out better way to rethrow errors from here.
+      // Adding `throws` keyword to the function results in different
+      // method signature in Objective-C. Maybe just call `RCTLogError`?
+      try? prop.set(value: value, onView: view)
+    } else if let _ = legacyViewManager {
```
Comment (tsapeta) [inline packages/expo-modules-core/ios/Fabric/ExpoFabricView.swift:63]: #### 🟠 SwiftLint: [Unused Optional Binding Violation](https://realm.github.io/SwiftLint/unused_optional_binding.html)
Prefer `!= nil` over `let _ =`

**Pull request merged**